hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -106,7 +106,8 @@ long getLastCallId() {
     public boolean register(Invocation invocation) {
         final long callId;
         try {
-            callId = callIdSequence.next(invocation.op.isUrgent());
+            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
```

Comment:
I am not sure how your scenario is possible. Non-retry operations are still amenable to max concurrent operations limitation. There can be at most (max concurrent non-retry operations + 1 retry) at any time. Can you elaborate your scenario?